src/cpp/src/text2image/stable_diffusion_3_pipeline.hpp

```diff
@@ -0,0 +1,576 @@
+            return generation_config.random_generator->next() * m_scheduler->get_init_noise_sigma();
+        });
+
+        // 6. Denoising loop
+        ov::Tensor noisy_residual_tensor(ov::element::f32, {});
+        ov::Tensor timestep;
+
+        for (size_t inference_step = 0; inference_step < generation_config.num_inference_steps; ++inference_step) {
+            // concat the same latent twice along a batch dimension in case of CFG
+            if (batch_size_multiplier > 1) {
+                batch_copy(latent, latent_cfg, 0, 0, generation_config.num_images_per_prompt);
+                batch_copy(latent,
+                           latent_cfg,
+                           0,
+                           generation_config.num_images_per_prompt,
+                           generation_config.num_images_per_prompt);
+
+                size_t timestep_size = generation_config.num_images_per_prompt * batch_size_multiplier;
+                timestep = ov::Tensor(ov::element::f32, {timestep_size});
+                float* timestep_data = timestep.data<float>();
+
+                for (size_t i = 0; i < timestep_size; ++i) {
+                    timestep_data[i] = timesteps[inference_step];
+                }
+            } else {
+                // just assign to save memory copy
+                latent_cfg = latent;
+                timestep = ov::Tensor(ov::element::f32, {1}, &timesteps[inference_step]);
+            }
+
+            ov::Tensor noise_pred_tensor = m_transformer->infer(latent_cfg, timestep);
+
+            ov::Shape noise_pred_shape = noise_pred_tensor.get_shape();
+            noise_pred_shape[0] /= batch_size_multiplier;
+            noisy_residual_tensor.set_shape(noise_pred_shape);
+
+            if (batch_size_multiplier > 1) {
+                // perform guidance
+                float* noisy_residual = noisy_residual_tensor.data<float>();
+                const float* noise_pred_uncond = noise_pred_tensor.data<const float>();
+                const float* noise_pred_text = noise_pred_uncond + noisy_residual_tensor.get_size();
+
+                for (size_t i = 0; i < noisy_residual_tensor.get_size(); ++i) {
+                    noisy_residual[i] = noise_pred_uncond[i] +
+                                        generation_config.guidance_scale * (noise_pred_text[i] - noise_pred_uncond[i]);
+                }
+            } else {
+                noisy_residual_tensor = noise_pred_tensor;
+            }
+
+            auto scheduler_step_result = m_scheduler->step(noisy_residual_tensor, latent, inference_step);
+            latent = scheduler_step_result["latent"];
+        }
+
+        float* latent_data = latent.data<float>();
+        for (size_t i = 0; i < latent.get_size(); ++i) {
+            latent_data[i] = (latent_data[i] / m_vae_decoder->get_config().scaling_factor) +
+                             m_vae_decoder->get_config().shift_factor;
```

Comment:
scaling factor is already applied as model post processing merged into model
https://github.com/openvinotoolkit/openvino.genai/blob/44cb9a90d827da10844fbb683d512c0da3e45cc6/src/cpp/src/text2image/models/autoencoder_kl.cpp#L93-L116

I think we need to merge `m_vae_decoder->get_config().shift_factor` into model as well. Let's just check that it's not `0.0f`

Reply:
discussed - it's SD3 specific code and should not be part of AutoencoderKL class